staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -366,19 +388,6 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 
 		w, err := r.listerWatcher.Watch(options)
 		if err != nil {
-			switch {
-			case isExpiredError(err):
-				// Don't set LastSyncResourceVersionExpired - LIST call with ResourceVersion=RV already
-				// has a semantic that it returns data at least as fresh as provided RV.
-				// So first try to LIST with setting RV to resource version of last observed object.
-				klog.V(4).Infof("%s: watch of %v closed with: %v", r.name, r.expectedTypeName, err)
-			case err == io.EOF:
-				// watch closed normally
-			case err == io.ErrUnexpectedEOF:
-				klog.V(1).Infof("%s: Watch for %v closed with unexpected EOF: %v", r.name, r.expectedTypeName, err)
-			default:
-				utilruntime.HandleError(fmt.Errorf("%s: Failed to watch %v: %v", r.name, r.expectedTypeName, err))
```

Comment:
This is the only case that is abnormal. If the goal is to expose unhandled internal errors, why would we forward expected errors to a custom handler?

Reply:
Your suggestion is what I did in the original version of this PR:
https://github.com/kubernetes/kubernetes/commit/13f9ad7cf008eba732157603701118a7997cf3cc

See discussion further up in the PR. I don't have a well-informed opinion on what the right answer is.